### PR TITLE
Add logs to api removeVpnUser

### PR DESCRIPTION
**Pull request by GutoVeronezi**
### Description
We verified that sometimes the command `removeVpnUser` fails, but we can't find the reason because there is no clear log message.

This PR intends to improve logging by adding the stack trace in case of fail of `removeVpnUser` command.

### Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functioanlity)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale
- [ ] Major
- [x] Minor

### How Has This Been Tested?
It has been tested locally on a test lab, calling the api.

### CODE REVIEW

**Comment by weizhouapache** [comment]
I am more interested the reason and the fix. do you have clue ?

**Comment by DaanHoogland** [comment]
I agree with @weizhouapache , @GutoVeronezi . Did you implement this locally and did you find a root cause?

**Comment by GutoVeronezi** [comment]
I have tested locally and couldn't found the root cause yet. Anyway, there is a lack of logs in this process.

**Comment by DaanHoogland** [comment]
hope you can build locally because before this is in a release we are all a few months older ;)

**Comment by rafaelweingartner** [comment]
Guys, this is an interesting improvement in logging. Sadly, this has not been reproduced anymore. That is why we proposed an improvement for the logging here, to help others to better troubleshoot this if it ever happens with other people.

**Comment by GabrielBrascher** [comment]
Is this a known issue? Can anyone else attest/reproduce it?

I am not a VPN user, the last time I have played with it was a long time ago. What I can say is that I agree with you all.

@DaanHoogland @weizhouapache I totally agree with the preference for looking into a fix.

However, on the other hand, it seems to be a corner case and the contributor had no conditions to catch a propper log neither to reproduce it. Improving logs seems logical to me as well as a very necessary first step.

It can be really tricky to debug/reproduce issues and sometimes we get into a limbo where we cannot properly debug nor monitor issues. Is this the case @rafaelweingartner @GutoVeronezi?

File: api/src/main/java/org/apache/cloudstack/api/command/user/vpn/RemoveVpnUserCmd.java [hunk 1]
```diff
@@ -110,19 +111,31 @@ public String getEventType() {
...
+            appliedVpnUsers = _ravService.applyVpnUsers(ownerId, userName);
+        } catch (ResourceUnavailableException ex) {
+            String errorMessage = String.format("Failed to refresh VPN user=[%s] due to resource unavailable. VPN owner id=[%s].", userName, ownerId);
+            s_logger.error(errorMessage, ex);
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, errorMessage, ex);
+        }
+
+        if (!appliedVpnUsers) {
+            String errorMessage = String.format("Failed to refresh VPN user=[%s]. VPN owner id=[%s].", userName, ownerId);
+            s_logger.debug(errorMessage);
```
Comment (DaanHoogland) [inline api/src/main/java/org/apache/cloudstack/api/command/user/vpn/RemoveVpnUserCmd.java:134]:
```suggestion
            if(s_logger.isDebugEnabled()) {
                String errorMessage = String.format("Failed to refresh VPN user=[%s]. VPN owner id=[%s].", userName, ownerId);
                s_logger.debug(errorMessage);
            }
```

File: server/src/main/java/com/cloud/network/vpn/RemoteAccessVpnManagerImpl.java [hunk 2]
```diff
@@ -419,7 +419,9 @@ public VpnUser doInTransaction(TransactionStatus status) {
     public boolean removeVpnUser(long vpnOwnerId, String username, Account caller) {
         final VpnUserVO user = _vpnUsersDao.findByAccountAndUsername(vpnOwnerId, username);
         if (user == null) {
-            throw new InvalidParameterValueException("Could not find vpn user " + username);
+            String errorMessage = String.format("Could not find VPN user=[%s]. VPN owner id=[%s]", username, vpnOwnerId);
+            s_logger.debug(errorMessage);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/vpn/RemoteAccessVpnManagerImpl.java:423]:
```suggestion
            if(s_logger.isDebugEnabled()) {
                String errorMessage = String.format("Could not find VPN user=[%s]. VPN owner id=[%s]", username, vpnOwnerId);
                s_logger.debug(errorMessage);
            }
```

Comment (GutoVeronezi) [reply server/src/main/java/com/cloud/network/vpn/RemoteAccessVpnManagerImpl.java:423]: Or we could update `log4j` to version 2. :grin:

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/network/vpn/RemoteAccessVpnManagerImpl.java:423]: yes, please. to 3 by now. but as we don't use facadeproxy modules for 3rd party libraries that's not a trivial job and needs heavy sponsoring by volunteers :togueincheek: :a.k.a.: :p

**Comment by GutoVeronezi** [comment]
@DaanHoogland @GabrielBrascher @weizhouapache until now we did not reproduce the mentioned error; However, improve logging may help others to better troubleshoot this if it ever happens. Any update about this PR?

**Review by weizhouapache — APPROVED** [review]
code lgtm
did not test it

**Review by GabrielBrascher — APPROVED** [review]
Sounds good to me; I understand that there are some corner-case situations that are hard to reproduce and debug and having this log might help when it happens again.

Code LGTM.

**Review by sureshanaparti — APPROVED** [review]
logs changes LGTM

**Comment by GutoVeronezi** [comment]
@DaanHoogland @GabrielBrascher @weizhouapache @sureshanaparti is there anything else to do?

**Comment by rhtyd** [comment]
Yes there are no tests
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Comment by GutoVeronezi** [comment]
@DaanHoogland @GabrielBrascher @weizhouapache @sureshanaparti @rhtyd is there anything else to do?

**Comment by GutoVeronezi** [comment]
@DaanHoogland @GabrielBrascher @weizhouapache @sureshanaparti @rhtyd is there anything else to do?

**Comment by nvazquez** [comment]
@blueorangutan package

**Pull request merged**